### ISSUE TITLE
Basic demos do not seem to work

### ISSUE THREAD

**Issue by konsumer** [replit/kaboom#108]
I can't seem to get any of the most basic demos to work. I tested on latest Chrome on OSX & Ubuntu 20.10. It seems to be to be a bad build or something.

First I tried this, from the front page:

```html
<script src="https://kaboomjs.com/lib/0.5.1/kaboom.js"></script>
<script type="module">
// initialize kaboom context
const k = kaboom()

// add a text of size 32 at position (100, 100)
k.add([
    k.text("oh hi", 32),
    k.pos(100, 100)
])
</script>
```

I get a blank play-area, but no text. Error-console says `Uncaught TypeError: Cannot read property 'lastObjID' of undefined`

To make sure it wasn't some plugin interference, I tried in incognito. Same problem.

I am a web-dev, so I am used to running a  dev-server (I like vite) and all that, so I setup a simple project:

```sh
npm i -D vite
npm i kaboom
touch index.html
vite
```

and made a  index.html, with only this:

```html
<script type="module">
import kaboom from 'kaboom'

// initialize kaboom context
const k = kaboom()

// add a text of size 32 at position (100, 100)
k.add([
  k.text('oh hi', 32),
  k.pos(100, 100)
])
</script>
```

I get `Uncaught TypeError: kaboom is not a function`

I also tried

```js
import * as kaboom from 'kaboom'
console.log(kaboom)
```

And it's an empty object. I thought maybe it's monkey-patching global `kaboom`, as an import side-effect, so I should try not naming the import the same thing:

```js
import * as k from 'kaboom'
console.log(kaboom)
```

and I got `undefined`

Am I doing something wrong or missing a step?

**Comment by slmjkdbtl** [comment]
Sorry for the trouble

- The example on kaboomjs.com is an example of v0.6 which will be released very soon, the `0.5.1` version tag is automatically generated with the last released version, so it's wrong right now. In v0.6 it's not required to use a `scene` but v0.5.1 does, thus the error
- v0.5.1 is known to have problems with some bundlers (#75), should be fixed by the v0.6 release soon

**Comment by konsumer** [comment]
> Sorry for the trouble

No prob! I understand you're still working on it. Just wanted to alert you if it's expected to work, as it doesn't work as in the docs. Also, I want to play with it, because it looks cool!

> The example on kaboomjs.com is an example of v0.6 which will be released very soon, the 0.5.1 version tag is automatically generated with the last released version, so it's wrong right now. In v0.6 it's not required to use a scene but v0.5.1 does, thus the error (if you change the script src to https://kaboomjs.com/lib/dev/kaboom.js it'll work)

Aha! Thanks! That got me started. Maybe the autobuild should be disabled for that part of the docs? I evaluate a lot of tools/libraries very quickly sometimes, and when the project-stated demo doesn't work out of the box, I generally just move on to something else (often reporting the error, so the maintainer knows it didn't work for me.)

> v0.5.1 is known to have problems with some bundlers ('default' is not exported by kaboom.js #75), should be fixed by the v0.6 release soon

Yeh, I am not using default like this:

```js
import kaboom from 'kaboom'
```

specifically to address this. I am illustrating it doesn't export anything at all (not `default`, or any other members.)

I wonder if [microbundle](https://github.com/developit/microbundle) might simplify the build. I have had really good experience with low-config setup, and it just works immediately for building cjs, mjs, umd, and more. LIie essentially just add htis to package.json:

```js
{
  "name": "kaboom",
  "type": "module",
  "source": "src/kaboom.js",
  "exports": "./dist/kaboom.modern.js",
  "main": "./dist/kaboom.cjs",
  "module": "./dist/kaboom.module.js",
  "unpkg": "./dist/kaboom.umd.js",
  "scripts": {
    "build": "microbundle"
  }
}
```

and gitignore dist. I also like all the sourcemaps it generates to `files`, but is optional.

**Comment by slmjkdbtl** [comment]
Kaboom is using `esbuild` now which doesn't have problem with building for multiple targets, the only problem was for user bundlers to recognize the settings in kaboom's `package.json` and choose the right file for the right target. In `v0.5.1` there's no `main` and `module` field in `package.json` thus it doesn't work with some bundlers (they always pick up `browser` field and use the iife target)
